transport/client.go

```diff
@@ -311,7 +315,7 @@ func Dial(address string, tr *Transport) (*tls.Conn, error) {
 
 	for _, chain := range state.VerifiedChains {
 		for _, cert := range chain {
-			revoked, ok := revoke.VerifyCertificate(cert)
```

Comment:
My idea is we still keep the original methods around: declare an unexported package variable `defaultChecker` with hardfail=true and a empty map, run those package level methods through this defaultChecker. So you don't need to update everywhere revoke is used.